### DISCUSSION TITLE
PWA manifest.json in extrahead

### DISCUSSION THREAD

**Discussion by AhmedThahir** [squidfunk/mkdocs-material#5263] [Q&A] [answered]
I want to add manifest.json to make the site into a progressive web app, but i'm not able to add it into head tag.

https://github.com/AhmedThahir/Uni_Notes/blob/main/mkdocs.yml#L18

Could you help me with this.

**Comment by squidfunk** [comment, marked as answer]
`extrahead` is a block, not a configurations setting. Use [theme extension](https://squidfunk.github.io/mkdocs-material/customization/#overriding-blocks) to add custom meta tags in a custom `main.html`:

```
{% extends "base.html" %}

{% block extrahead %}
  ...
{% endblock %}

```

**Comment by AhmedThahir** [comment]
https://github.com/AhmedThahir/Uni_Notes/blob/d29aaef587c57630ae4a08a77b76e18b0f09e327/overrides/main.html#L1-L5

I've done as explained in the documentation, but now my question is. Is there a way to make all html files link to the same location?

For example, `manifest.json` is in the `docs`. I want all folders in the current directory as well as in sub-directories to be able to access it.

**Reply by kamilkrzyskow** [reply]
```html
<link rel="manifest" href="./manifest.json">
```
should probably be
```html
<link rel="manifest" href="{{ 'manifest.json' | url }}">
```

Based on examples:
https://github.com/squidfunk/mkdocs-material/blob/master/src/base.html#L86
https://github.com/mkdocs/mkdocs/blob/master/mkdocs/themes/mkdocs/base.html#L19-L29

The ` | url ` pipes the data on the left to the `url` filter function provided by MkDocs:
https://github.com/mkdocs/mkdocs/blob/56b235a8ad43f2300d17f87e6fa4de7a3d764397/mkdocs/utils/filters.py#L12

**Reply by squidfunk** [reply]
That's correct, you must use the `url` filter for all assets since MkDocs will build relative paths.
